boot/bootutil/src/bootutil_misc.c

```diff
@@ -157,7 +157,7 @@ boot_magic_off(const struct flash_area *fap)
 static inline uint32_t
 boot_image_ok_off(const struct flash_area *fap)
 {
-    return boot_magic_off(fap) - BOOT_MAX_ALIGN;
+    return (boot_magic_off(fap) - BOOT_MAX_ALIGN) & ~(BOOT_MAX_ALIGN - 1);
```

Comment:
Should we apply this same patch to  https://github.com/mcu-tools/mcuboot/blob/main/boot/bootutil/src/bootutil_public.c#L143?